Cookbook/Technical-Documents/BackendFeatureHowTo.md

```diff
@@ -0,0 +1,204 @@
+
+``` swift
+func access<Request, Response>(_ resource: BackendResource<Request, Response, ConcreteAuth>, with request: Request) -> SignalProducer<Response, CoreError>
+```
+
+Using either `Accessible` or `AuthenticatedAccessible`, you can perform the network request defined by a backend resource, and this is exactly what we do in the business controller.
+
+### Business controller
+A `BusinessController` is the entity you will be interacting with in the view model, and as such, it needs to be injected in order to be stubbed for testing the view model, which means it is defined first as a protocol:
+
+``` swift
+protocol CitiesBusinessControllerProtocol {
+    func fetch() -> SignalProducer<CitiesResponse, CoreError>
+}
+```
+
+Again, the business controller uses `fetch` terminology for getting some data from the backend. As this is the default and only method which gets no parameters, we call it simply `fetch()` without any modifiers.
+
+It returns a signal with the parsed response, which we can use in our view model directly.
+
+Let's define the concrete class for this protocol:
+``` swift
+struct CitiesBusinessController: CitiesBusinessControllerProtocol {
+    private let accessible: AuthenticatedAccessible
+
+    init(accessible: AuthenticatedAccessible) {
+        self.accessible = accessible
+    }
+
+    func fetch() -> SignalProducer<CitiesResponse, CoreError> {
+        let resource = CitiesService.fetchCitiesResource()
+        return accessible.access(resource)
+    }
+}
+```
+
+As simple as that: we get a backend resource from `CitiesService` , access it with the provided `AuthenticatedAccessible` and return the result.
+
+Note that we didn't specify the second parameter (request) for `accessible.access(…)`. Earlier we defined the request type as `Void`, which allows us to use a single-parameter convenience method on `AuthenticatedAccessible`, it simply calls `access(resource, with: ())`.
+
+Let's see how we may use the business controller in a view model.
+
+### View model
+In the view model, we usually keep the results of a fetch in the `State`:
+``` swift
+struct State {
+    var cities: [City]
+}
+```
+
+The entities we're loading are usually supplied through an `Event`:
+``` swift
+enum Event {
+    case didLoad(CitiesResponse)
+}
+```
+
+When the view model gets an event, `reduce` is called and we can set the new value:
+``` swift
+static func reduce(state: State, event: Event) -> State {
+    switch event {
+    case let .didLoad(citiesResponse):
+        return state.with(set(\.cities, citiesResponse.cities))
+    }
+}
+```
+
+Now `state.cities` contains the values we fetched and parsed from the network!
+
+In order to trigger loading, we define a `Feedback` and transform a network response in order to conform to the signature of the effect:
+``` swift
+{ _ -> SignalProducer<Event, NoError> in
```

Comment:
can you also put the function signature/definition? 
it just feels odd to not have it.

Reply:
Yeah, this was extracted from a convoluted example, I think I can define a super simple feedback here and have its full signature ✨

Reply:
Agree, we should have a function like `whenLoading(cities:)` which aligns with our conventions.